src/aap_eda/services/project/imports.py

```diff
@@ -46,27 +45,39 @@ class ProjectImportError(Exception):
     pass
 
 
+class MalformedError(Exception):
+    pass
+
+
 def _project_import_wrapper(
     func: Callable[[ProjectImportService, models.Project], None]
 ):
     @wraps(func)
     def wrapper(self: ProjectImportService, project: models.Project):
         project.import_state = models.Project.ImportState.RUNNING
-        project.save()
+        project.save(update_fields=["import_state"])
+        error = None
         try:
             func(self, project)
             project.import_state = models.Project.ImportState.COMPLETED
-            project.save()
-        except (utils.IntegrityError, exceptions.ObjectDoesNotExist):
-            logger.exception("Object may have been deleted")
         except Exception as e:
+            project.import_state = models.Project.ImportState.FAILED
+            project.import_error = str(e)
+            error = e
+        finally:
             try:
-                project.import_state = models.Project.ImportState.FAILED
-                project.import_error = str(e)
-                project.save()
+                project.save(update_fields=["import_state", "import_error"])
             except exceptions.ObjectDoesNotExist:
-                logger.exception("Project may have been deleted")
-            raise
+                raise ProjectImportError(
+                    "Project may have been deleted"
+                ) from error
+            else:
+                if error and isinstance(error, ProjectImportError):
+                    raise error
+                elif error:
+                    raise ProjectImportError(
+                        "Failed to import the project"
```

Comment:
This message might be captured by the upper try block and exposed to the user in the import_error field, so it would be a message without much meaning. I think it should include the original error message as well. What do you think?